christfellowshipchurch/apollos-api
# [Chat] Add channelType to StreamChatType schema

# Changes
Currently, the client apps have a hard coded `channelType` value used to initiate a chat channel.
Events use `"livestream"`, groups use `"group"`.

This PR lifts the resolution of that value for various chat contexts to the api, as `streamChatChannel.channelType`.

Types like `GroupItem`, `LiveStream`, etc can resolve the value independently for their specific context.

# Test Instructions
1. Open graphql playground locally
1. Run the following queries and verify that the `Group` query returns `"GROUP"`, and `LiveStream` returns `"LIVESTREAM"`

**Group**
```graphql
query getGroup {
  node(id: "Group:68bb3bf83c14c184a188bbdb8d984e94") {
    __typename
    ... on Group {
      id
      title
      streamChatChannel {
        id
        channelId
        channelType
      }
    }
  }
}
```

**LiveStream Event**
⚠️ _NOTE_: Will not work until we re-setup the livestream test events!
```graphql
query getEvents {
  allEvents {
    ... on EventContentItem {
      __typename
      id
      title
      summary
      liveStream {
        id
        isLive
        eventStartTime
        eventEndTime
        streamChatChannel {
          id
          channelId
          channelType
        }
      }
    }
  }
}
```
